viewing_party/party.py

```diff
@@ -1,23 +1,248 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title == None:
+        return None
+    elif genre == None:
+        return None
+    elif rating == None:
+        return None

```

Comment:
Nice handling for returning `None` for missing data! We could simplify this if-statment tree by using `and` or `or`:

```suggestion
    if title is None or genre is None or rating is None:
        return None
```
or
```
    if not (title and genre and rating):
        return None
```

I also want to touch on how we're comparing variables to `None`, here, we would want to use `is` rather than `==` for comparison because we are checking for `None`.

When comparing things in python, `==` compares if the values are equivalent, while `is` checks if they have the same identity. The `==` operator for objects can also be overloaded (re-defined) while `None` is a "Singleton object", an object which only ever has one instance, and thus one ID. Since there is only one instance of `None` allocated by python, all variables holding None must be pointing to the same ID in memory. All this put together means:
- When comparing a variable against `== None` could give us the wrong value.
- We want to use `is`, because if a variable holds None, it always points to the same ID as the None singleton object we refer to when we type out `None`.

Something else we can consider: here we're explicitly checking if the values are None, but what if we got empty strings for one of the values? If we wanted to check for those at the same time as we check for None, we could rely on the truthy/falsy values of the parameters and write something like: `if not title or not genre or not rating`:

Reply:
Thank you for this! I think I'm still a little uncertain about when I can use stuff like not and combine things - I personally like seeing all the details and/or potential routes of something laid out, but that isn't necessarily best practice!

Reply:
And I hadn't realized that about None at all, so that's good to know as well.